src/rust/ide/view/graph-editor/src/component/node/output/port.rs

```diff
@@ -434,8 +456,33 @@ impl Model {
             eval opacity.value ((t) shape.set_opacity(*t));
 
 
+            // === Label Opacity ===
+
+            label_visibility        <- frp.on_hover.and(&frp.set_label_visibility);
+            label_opacity.target    <+ label_visibility.on_true().constant(PORT_OPACITY_HOVERED);
+            label_opacity.target    <+ label_visibility.on_false().constant(0.0);
+            label_color             <- all_with(&color.value,&label_opacity.value,
+                |color,&opacity| color.opaque.with_alpha(opacity).into());
+            label.set_color_all     <+ label_color;
+            label.set_default_color <+ label_color;
+
+
             // === Size ===
 
+            frp.source.size <+ frp.set_size;
+            eval frp.size ((&s)
+                shape.set_size(s + Vector2(HOVER_AREA_PADDING,HOVER_AREA_PADDING) * 2.0));
+            set_label_x <- all_with(&frp.size,&label.width,
+                f!([port_count,port_index](port_size,label_width) {
+                    let corner_circumference  = node::RADIUS * 2.0 * PI;
+                    let corner_segment_length = corner_circumference * 0.25;
+                    let center_segment_length = port_size.x - node::RADIUS * 2.0;
+                    let shape_border_length   = center_segment_length + corner_segment_length * 2.0;
+                    let relative_x = (port_index as f32 + 0.5) / (port_count as f32) - 0.5;
+                    let center_x = relative_x * shape_border_length;
+                    center_x - label_width/2.0
+                }));
```

Comment:
This part duplicates some logic from the shape definitions, but it seems difficult to avoid that.

Reply:
Please talk with Michael, who was implementing the existing equations if reusing them could be possible. Maybe we should not use `Var<f32>` in the equations but just `f32`? I mean - this might be possible, depending on usage. Please confirm this with Michael - for me, it seems like refactoring this logic should be doable, as we are really computing the same things there.